### PR TITLE
Introduce clearable datepicker

### PR DESCRIPTION
**Pull request by rymeskar**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Introduce clearable datepicker

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/docs/src/examples/components/Datepicker/States/DatepickerExampleClearable.shorthand.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { Datepicker } from '@fluentui/react-northstar';
+import * as React from 'react';
+
+const DatepickerExampleClearable = () => <Datepicker clearable={true} today={new Date(2020, 9, 1, 0, 0, 0, 0)} />;
```
Comment (pompomon) [inline packages/fluentui/docs/src/examples/components/Datepicker/States/DatepickerExampleClearable.shorthand.tsx:4]:
```suggestion
const DatepickerExampleClearable = () => <Datepicker input={{clearable: true}} today={new Date(2020, 9, 1, 0, 0, 0, 0)} />;
```
I was thinking more of just passing `clearable` to the `input` slot, so we don't need to forward the props

File: packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx [hunk 2]
```diff
@@ -231,7 +234,7 @@ export const Datepicker: ComponentWithAs<'div', DatepickerProps> &
       const futureError = validateDate(parsedDate, target.value, calendarOptions, dateFormatting, props.required);
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx:234]:
```suggestion
      const validationError = validateDate(parsedDate, target.value, calendarOptions, dateFormatting, props.required);
```
_nit_

File: packages/fluentui/docs/src/examples/components/Datepicker/States/DatepickerExampleClearable.shorthand.tsx (hunk 1 above)
Comment (rymeskar) [reply packages/fluentui/docs/src/examples/components/Datepicker/States/DatepickerExampleClearable.shorthand.tsx:4]: Makes sense. 

For `required` and `disabled`, I think we keep it explicit on the DatepickerProps, right?

Comment (pompomon) [reply packages/fluentui/docs/src/examples/components/Datepicker/States/DatepickerExampleClearable.shorthand.tsx:4]: Yeah, those are really common usage, so I would keep them as is :)

**Review by pompomon — APPROVED** [review]

File: packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx (hunk 2 above)
Comment (rymeskar) [inline packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx:234]: Refactor branching.

**Pull request closed without merging**